include/ofi_mr.h

```diff
@@ -160,7 +162,8 @@ struct ofi_mem_monitor {
 	 * to be re-registered.
 	 */
 	bool (*valid)(struct ofi_mem_monitor *notifier, const void *addr,
-		      size_t len, union ofi_mr_hmem_info *hmem_info);
+		      size_t len, void **ipc_handle,
```

Comment:
Right, if we can change `valid` API to valid(struct ofi_mr_info *info, struct ofi_mr_entry *entry)`. In that case we do not need to introduce `ipc_handle` to `ofi_mr_hmem_info`. We also do not need to modify the `subscribe` API. The only concern I have is such change makes `valid` quite inconsistent with other APIs in `ofi_mem_monitor`. But I do think it could be better.

Reply:
I updated the PR with this valid API change.